src/opam_create.ml

```diff
@@ -68,6 +68,28 @@ let package_fields
   optional @ dep_fields
 
 let opam_fields project (package : Package.t) =
+  let dune_version = Dune_project.dune_version project in
+  let dune_name = Package.Name.of_string "dune" in
+  let package =
+    if dune_version < (1, 11) || Package.Name.equal package.name dune_name then
+      package
+    else
+      let dune_dep =
+        let dune_version = Syntax.Version.to_string dune_version in
+        let constraint_ : Package.Dependency.Constraint.t =
+          Uop (Gte, QVar dune_version) in
+        { Package.Dependency.
+          name = dune_name
+        ; constraint_ = Some constraint_
+        }
+      in
+      let is_dune_depend (pkg : Package.Dependency.t) =
+        Package.Name.equal pkg.name dune_dep.name in
+      if List.exists package.depends ~f:is_dune_depend then
+        package
```

Comment:
Does it make sense to check if the constraint is correct?

Reply:
I figured that if there is a manual override, then there is a good reason for it (e.g. perhaps to exclude a particular dune version). Checking those isn't easy, so it's best to leave a human-written one as-is.